app/models/katello/host/content_facet.rb

```diff
@@ -31,11 +36,84 @@ class ContentFacet < Katello::Model
       has_many :content_facet_applicable_module_streams, :class_name => "Katello::ContentFacetApplicableModuleStream", :dependent => :delete_all, :inverse_of => :content_facet
       has_many :applicable_module_streams, :through => :content_facet_applicable_module_streams, :class_name => "Katello::ModuleStream", :source => :module_stream
 
-      validates :content_view, :presence => true, :allow_blank => false
-      validates :lifecycle_environment, :presence => true, :allow_blank => false
       validates_with ::AssociationExistsValidator, attributes: [:content_source]
+      validates_with Katello::Validators::GeneratedContentViewValidator
       validates :host, :presence => true, :allow_blank => false
-      validates_with Validators::ContentViewEnvironmentValidator
+
+      attr_accessor :cves_changed
+
+      def initialize(*args)
+        init_args = args.first || {}
+        env_id = init_args.delete(:lifecycle_environment_id)
+        cv_id = init_args.delete(:content_view_id)
+        super(*args)
+        if env_id && cv_id
+          assign_single_environment(
+            lifecycle_environment_id: env_id,
+            content_view_id: cv_id
+          )
+        end
+        self.cves_changed = false
+      end
+
+      def mark_cves_changed(_cve)
+        self.cves_changed = true
+      end
+
+      def cves_changed?
+        cves_changed
+      end
+
+      def multi_content_view_environment?
+        content_view_environments.size > 1
+      end
+
+      def single_content_view_environment?
+        content_view_environments.size == 1
+      end
+
+      def single_content_view
+        if multi_content_view_environment?
+          Rails.logger.warn _("Content facet for host %s has more than one content view. Use #content_views instead.") % host.name
+        end
+        content_view_environments.first.content_view
+      end
+
+      def single_lifecycle_environment
```

Comment:
Any reason not to just call it `def lifecycle_environment`?

That would fix the "Host - Applicable Errata" report, which is currently failing:
```
Association named 'lifecycle_environment' was not found on Host::Managed; perhaps you misspelled it? Did you mean? lifecycle_environments content_view_environments content_view_environment_content_facets applicable_rpms
```

Reply:
Subscription - Entitlement Report is also broken on the same issue.

Reply:
The default kickstart provisioning template is also affected: app/views/unattended/provisioning_templates/provision/kickstart_default.erb

Reply:
This was on purpose. The idea was that once we actually ramp up support for multi CV, it'll be that much easier to grep all our code for `single_lifecycle_environment` and see where we need to change. Thoughts?

I was planning on fixing the reports to use `single_`; just hadn't gotten to it yet.